### ISSUE TITLE
Request textDocument/documentSymbol failed. Error: name must not be falsy

### ISSUE THREAD

**Issue by goyalyashpal** [latex-lsp/texlab#941]
* Similar to #870
* except that i have no clue what is causing it

### Versions

* TexLab extension: v5.10.1
* VS Codium: v1.82.2
* OS: win 10 22H2

### Log

```log
[Error - 5:11:42 PM] Request textDocument/documentSymbol failed.
Error: name must not be falsy
	at s.validate (c:\Program Files\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:103:19664)
	at new s (c:\Program Files\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:103:19959)
	at V (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:213145)
	at c (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:251142)
	at t.map (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:251228)
	at Object.asDocumentSymbols (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:220963)
	at i (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:159617)
	at async s.provideDocumentSymbols (c:\Program Files\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:109:46486)
[Error - 5:11:43 PM] Request textDocument/documentSymbol failed.
Error: name must not be falsy
	at s.validate (c:\Program Files\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:103:19664)
	at new s (c:\Program Files\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:103:19959)
	at V (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:213145)
	at c (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:251142)
	at t.map (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:251228)
	at Object.asDocumentSymbols (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:220963)
	at i (d:\StandalonePrograms\vsc\vsc_ext\efoerster.texlab-5.10.1-win32-x64\dist\extension.js:1:159617)
	at async s.provideDocumentSymbols (c:\Program Files\VSCodium\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:109:46486)

```

**Comment by goyalyashpal** [comment]
this error is common in vs code world: 

* https://github.com/Microsoft/vscode/issues/15018 <-- https://stackoverflow.com/questions/40549539/40631794#40631794
* https://github.com/Microsoft/vscode-cpptools/issues/1373
* https://github.com/vuejs/language-tools/issues/2438
* https://github.com/Dart-Code/Dart-Code/issues/3738

**Comment by pfoerster** [comment]
@goyalyashpal Thanks for the report! Do you have a reproducing example that you can share? It is odd that this error still appears after explicitly filtering out symbols without a name (e. g. empty `\section`s)

**Comment by goyalyashpal** [comment]
it's not happening now...

the issue is not there anymore.
* i think removing the aux files and configuring them to go to a separate directory fixed it
* anyhow, i have uninstalled and reinstalled texlab extension, and this issue is not there anymore.

**Comment by goyalyashpal** [comment]
it's not happening now... the issue is not there anymore.

i did lots of things between filing this issue and till now, so, not sure what was it. i think removing the aux files and configuring them to go to a separate directory (in texstudio) was what fixed it

**Comment by goyalyashpal** [comment]
oh well well, it manifested again 🤦 


https://github.com/latex-lsp/texlab/assets/19423063/4122fbe6-27e9-4182-9ee2-dca316eb34da

**Comment by pfoerster** [comment]
@goyalyashpal Can you share the `chapter.tex` file (or any other file that triggers the issue)? The part shown in the video is not unfortunately not enough to reproduce the issue.

**Comment by goyalyashpal** [comment]
hi again! yeah, i was thinking of trying to make a minimal example for the bug, but couldn't get time. but thanks for reminder - i will try to make it in some days.

**Comment by goyalyashpal** [comment]
hi! do you have any hunch of what might be causing the problem? maybe it would prove useful in giving me some direction towards making a minimal repro example

like _any_ near/far idea as to what part to try to keep and what to remove to narrow down
or like any past incident or experience with such issue

**Comment by goyalyashpal** [comment]
something / anything

**Comment by pfoerster** [comment]
> like any near/far idea as to what part to try to keep and what to remove to narrow down
or like any past incident or experience with such issue

The document symbols implementation only considers the following constructs:
- `\section`-like commands (including `\item`)
- Environments
- BibTeX entries and fields

So you should be able to remove unrelated commands and text passages.

**Comment by pfoerster** [comment]
@goyalyashpal Probably, this is a duplicate of #1040 and should be fixed with the new `v5.14.0` release.
